### DISCUSSION TITLE
HX711 load cell

### DISCUSSION THREAD

**Discussion by dittmanndennis** [micropython/micropython#12030] [Hardware & Peripherals]
Hi, I just started up my load cell connected through a HX711 to a ESP32. I am using the [hx711_gpio code](https://github.com/robert-hh/hx711/blob/master/hx711_gpio.py) from @robert-hh to interact with the data from the sensor. This is all working well. However, the readings that I receive, no matter what weight I put on my sensor, vary around 1910 ±25 (this is the raw data from the read() function).

I wonder why the sensor does not respond to weight. Could it be anything else but a faulty sensor? Should I maybe redo the soldering? What else could it be?

Schematics (I followed this tutorial, should be standard): https://randomnerdtutorials.com/esp32-load-cell-hx711/

**Comment by robert-hh** [comment]
Most likely this  behavior is caused by a wrong connection of the load cell to the HX711 board. Please note, that depending on the selected gain either the A or B inputs are used. For gain=128 and gain=64 it's input A, for gain=32 it's input B.

**Reply by dittmanndennis** [reply]
I'll try to redo the soldering over the weekend. Thanks for the input so far!

**Comment by synapticonashaikh** [comment]
To check if the sensor or the connection is proper, I suggest running the basic sample code provided by Arduino IDe (in C++). The sample code (stable version) is available in the "add library" sections. Some random tutorials are as follows:
1. https://randomnerdtutorials.com/esp32-load-cell-hx711/
2. https://www.electroniclinic.com/esp32-hx711-and-load-cell-based-digital-weighing-scale-iot-weighing-scale/

**Reply by scruss** [reply]
The OP may not have a board that the Arduino subsystem runs on. It's also just as likely that there will be problems with the Arduino code. There are mature libraries for the HX711 for MicroPython. Probably best to keep it in our ecosystem.

**Comment by bidrohini** [comment]
Hi, please check if you have done the calibration properly. Also, check with your multimeter if there had been an accidental short circuit somewhere.

**Reply by robert-hh** [reply]
@bidrohini If the value does not change, it's not a matter of calibration.

**Reply by bidrohini** [reply]
Short circuit can still be the issue, right?

**Reply by robert-hh** [reply]
Short circuit or other wiring problem could be an issue.
